DesignPatternSession2/src/com/metacube/shoppingCart/Dao/BaseDao.java

```diff
@@ -0,0 +1,36 @@
+package com.metacube.shoppingCart.Dao;
+
+import java.util.Map;
+
+import com.metacube.shoppingCart.Entity.BaseEntity;
+
+/**
+ * BaseDao interface extended by all Dao Interfaces
+ * 
+ * @author Udit Saxena
+ * @param <T>
+ *            generic type only of entity
+ */
+public interface BaseDao<T extends BaseEntity> {
```

Comment:
No need for have a common Dao as base, every entity can have its separate methods in dao